config_verifier/src/index.ts

```diff
@@ -0,0 +1,520 @@
+                
+                    targetPeerInformation['peerAddress'] = "0x" + bufferToHex(targetPeerInformationTmp['address'])
+                    targetPeerInformation['tokenDecimals'] = targetPeerInformationTmp['tokenDecimals'];
+                }
+                catch(e){ // Don't see peer on Solana
+                    targetPeerInformation["peerAddress"] = "0x000000000000000000000000000000000000000000000" // Can't find the addresss, since it doesn't exist                    
+                    targetPeerInformation['tokenDecimals'] = BigNumber.from(0);
+                }
+            }else{
+                console.log("Invalid chain type...");
+            }
+
+            if(BigNumber.from(targetPeerInformation['peerAddress']).eq(0)){
+                console.log(`ERROR: Peer on chain ${src_id} is not on chain ${target_id}`)
+                continue;
+            }
+
+            // Ensure that the peers match up between the target and destination
+            if(!BigNumber.from(targetPeerInformation['peerAddress']).eq(src['managerAddress'])){
+                console.log(`ERROR: Peers don't match chain id for manager sender chain ${target_id} and receiving chain ${src_id}`);
+            }
+
+            // Check that the decimals for the configuration match up with the other chains decimals
+            if(targetPeerInformation['tokenDecimals'] != src['tokenDecimals']){
+                console.log(`ERROR: Peers don't match decimals for sender chain ${target_id} with decimals ${targetPeerInformation['tokenDecimals']} and receiving chain ${src_id} with decimals ${src['tokenDecimals']}`);
+            }  
+
+            var inboundRateLimitParams; 
+            if(target['type'] == 'evm'){ 
+                // Inbound rate limit is sane
+                inboundRateLimitParams = await target['manager'].getInboundLimitParams(src['chainId'] );
+
+            }else if(target['type'] == 'solana'){
+                inboundRateLimitParams = (await target['manager'].program.account.inboxRateLimit.fetch(target['manager'].inboxRateLimitAccountAddress(src['chainId'])))['rateLimit'];
+            }
+            
+            // Rate limit is set to 0. Indicates that it's disabled.
+            if(inboundRateLimitParams['limit'].eq(0)){
+                console.log(`WARNING: Inbound rate limit disabled for sender chain ${target_id} and target chain ${src_id}`); 
+                continue;
+            }
+
+            // Don't feel this is necessary or proper. Depends on the use case.
+            // if(inboundRateLimitParams['limit'].eq(BigNumber.from(2).pow(64).sub(1))){
+            //     console.log(`WARNING: Inbound rate limit very high for ${target_id} and target chain ${src_id}`); 
+            // }
+        }
+    } 
+
+    // Checking if the mint/burn setup is correct
+    if(Object.keys(chainData).length -1 != burn || lock != 1){
```

Comment:
Isn't it possible to do only a mint-and-burn model without a locking chain? I thought this was part of the architecture at least at a certain point of the design discussions.

It might be good to add a check for this. If you don't want to complicate this PR then maybe we could add it as an Issue